src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/PiglinBarterListener.java

```diff
@@ -0,0 +1,76 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.bukkit.entity.Item;
+import org.bukkit.entity.Piglin;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDropItemEvent;
+import org.bukkit.event.entity.EntityPickupItemEvent;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.thebusybiscuit.slimefun4.core.attributes.NotPlaceable;
+import io.github.thebusybiscuit.slimefun4.core.attributes.RandomMobDrop;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+
+public class PiglinBarterListener implements Listener {
+
+    /**
+     * Listens to the EntityDropItemEvent event to inject a {@link RandomMobDrop} if its dropChance() check passes.
+     * this would only be possible if the {@link RecipeType} is of PiglinBarter type. This listener will only ever
+     * be enabled if the server version is 1.16 or above.
+     * 
+     * @author dNiym
+     * 
+     * @see getBarterDrops
+     * @see RandomMobDrop
+     * 
+     */
+    private static boolean hasPiglins = false;
+
+    public PiglinBarterListener(SlimefunPlugin plugin) {
+        
+        try {
+            Class.forName("org.bukkit.entity.Piglin");
```

Comment:
Check in the main class if is version before initialising. No need for this class call or boolean. MinecraftVersion has a isAtLeast